File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
def solution(bandage, health, attacks):
    state = health
    def heal(state, time):
        state += time * bandage[1]
        state += time//bandage[0] * bandage[2]
        if state > health:
            state = health
        return state
    def attack(state, damage):
        state -= damage
        if state <= 0:
            return 0
        return state
    answer = 0
    cur = 0
    
    for i in range(len(attacks)):
        sec = attacks[i][0] - cur - 1
        cur = attacks[i][0]
        state = heal(state, sec)
        state = attack(state, attacks[i][1])
        if not state:
            return -1
    return state

File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
        int cur = 0;
        int h = health;
        int sec;
        for (int r = 0; r < attacks.length; r++){
            sec = attacks[r][0] - cur;
            h += (sec-1) * bandage[1] + ((sec-1)/bandage[0] * bandage[2]);
            h = Math.min(h, health);
            h -= attacks[r][1];
            if (h <= 0)
                return -1;
            cur = attacks[r][0];
        }
        return h;
    }
}
    // [3, 2, 7]
	// 	[[1, 15], [5, 16], [8, 6]]